extensions/resteasy-reactive/quarkus-resteasy-reactive-links/runtime/src/main/java/io/quarkus/resteasy/reactive/links/runtime/PathParameterValueResolver.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.resteasy.reactive.links.runtime;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.Set;
+
+final class PathParameterValueResolver {
+
+    <T> Object[] resolveAll(Set<PathParameterInfo> pathParametersInfo, T instance) {
+        Set<Object> values = new HashSet<>(pathParametersInfo.size());
+        for (PathParameterInfo pathParameterInfo : pathParametersInfo) {
+            values.add(resolve(pathParameterInfo, instance));
+        }
+
+        return values.toArray(new Object[0]);
+    }
+
+    private <T> Object resolve(PathParameterInfo pathParameterInfo, T instance) {
+        for (Method method : instance.getClass().getMethods()) {
+            if (method.getName().equals(pathParameterInfo.getValueGetter())) {
+                try {
+                    return method.invoke(instance);
```

Comment:
We can get rid of this reflection by generating bytecode that invokes it for a given instance like this:

```java
public class $$Order$Id implements Getter<Order,Long> {
 public static Long get(Order order){
  return order.$$generated_getter_id();
 }
}
```

And you instantiate those getters at static init and can just invoke them without reflection when you need to.

Reply:
Nice, I'll add that.